### ISSUE TITLE
Remove `agent-groups send full` garbage collector workaround

### ISSUE THREAD

**Issue by GGP1** [wazuh/wazuh#21611]
## Description

In https://github.com/wazuh/wazuh/pull/16012, we introduced a workaround to avoid a bug in the CPython streams library that was causing the `agent-groups send full` tasks to be garbage collected before running. 

https://github.com/wazuh/wazuh/blob/6d59398bd3647d7cf07d9b8457b943fc5b28dc05/framework/wazuh/core/cluster/common.py#L131-L133

This was necessary in the `3.9.9` Python version, but a [fix](https://github.com/python/cpython/issues/90467) has been introduced in `3.10.8` and it is no longer needed.

This issue aims to remove the code that is no longer necessary, update the tests related and verify that the tasks are executed successfully.

## Checks

The following elements have been updated or reviewed (should also be checked if no modification is required):
- [x] Tests (unit tests, API integration tests).
- [x] Changelog.
- [x] Documentation.
- [x] Integration test mapping (using `api/test/integration/mapping/_test_mapping.py`).

**Comment by GGP1** [comment]
## Update

Removed the garbage collector workaround and tried to reproduce the error after the changes.

**Comment by GGP1** [comment]
## Update

As it has been discussed in https://github.com/python/cpython/issues/91887, `asyncio` only keeps a `weakref.WeakSet` of all tasks and it does not discriminate between done and pending tasks.

As described by the [weakref](https://docs.python.org/3/library/weakref.html) package documentation:

```
A weak reference to an object is not enough to keep the object alive: when the only remaining references to a referent are weak references, garbage collection is free to destroy the referent and reuse its memory for something else
```

Meaning that the task we are creating and assigning to `self.task` can be garbage collected at any point in time, and there is no reference preventing that from happening.

> [Here](https://github.com/python/cpython/issues/91887#issuecomment-1805959882) a user provides a snippet to replicate the issue in 3.10 that is similar to our use case.

We can see the tasks being garbage collected in the [logs](https://github.com/wazuh/wazuh/pull/21694#issue-2112771714) provided in the PR description. The task created in this case is `send_entire_agent_groups_information` and we can see that each task starts its execution but only one of them finishes. This is because only one task is kept in the heap memory while the others are garbage collected. 

https://github.com/wazuh/wazuh/blob/a4b03d6ab30fa65e40ae06d823283e59df3a8322/framework/wazuh/core/cluster/master.py#L653-L679

One of the proposed solutions is to introduce a set `_pending_tasks` to keep a hard-reference to all pending tasks, but it hasn't been decided whether this will be included in a future release or at all. 

To sum up, the issue persists in 3.10 and hasn't been fixed by https://github.com/python/cpython/issues/90467. We should keep the hard reference unchanged and update the comment to avoid any confusion in the future.

**Comment by GGP1** [comment]
## Update

As it has been discussed in https://github.com/python/cpython/issues/91887, `asyncio` only keeps a `weakref.WeakSet` of all tasks and it does not discriminate between done and pending tasks.

As described by the [weakref](https://docs.python.org/3/library/weakref.html) package documentation:

```
A weak reference to an object is not enough to keep the object alive: when the only remaining references to a referent are weak references, garbage collection is free to destroy the referent and reuse its memory for something else
```

Meaning that the task we are creating and assigning to `self.task` can be garbage collected at any point in time, and there is no reference preventing that from happening.

> [Here](https://github.com/python/cpython/issues/91887#issuecomment-1805959882) a user provides a snippet to replicate the issue in 3.10 that is similar to our use case.

We can see the tasks being garbage collected in the [logs](https://github.com/wazuh/wazuh/pull/21694#issue-2112771714) provided in the PR description. The task created in this case is `send_entire_agent_groups_information` and we can see that each task starts its execution but only one of them finishes. This is because only one task is kept in the heap memory while the others are garbage collected. 

https://github.com/wazuh/wazuh/blob/a4b03d6ab30fa65e40ae06d823283e59df3a8322/framework/wazuh/core/cluster/master.py#L653-L679

> [!Note]
> The asynchronous operations performed inside each task are correctly awaited, it shouldn't be an issue with how we handle asynchrony 

One of the proposed solutions is to introduce a set `_pending_tasks` to keep a hard-reference to all pending tasks, but it hasn't been decided whether this will be included in a future release or at all. 

To sum up, the issue persists in 3.10 and hasn't been fixed by https://github.com/python/cpython/issues/90467. We should keep the hard reference unchanged and update the comment to avoid any confusion in the future.

**Comment by GGP1** [comment]
## Update

As it has been discussed in https://github.com/python/cpython/issues/91887, `asyncio` only keeps a `weakref.WeakSet` of all tasks and it does not discriminate between done and pending tasks.

As described by the [weakref](https://docs.python.org/3/library/weakref.html) package documentation:

```
A weak reference to an object is not enough to keep the object alive: when the only remaining references to a referent are weak references, garbage collection is free to destroy the referent and reuse its memory for something else
```

Meaning that the task we are creating and assigning to `self.task` can be garbage collected at any point in time, and there is no reference preventing that from happening.

> [Here](https://github.com/python/cpython/issues/91887#issuecomment-1805959882) a user provides a snippet to replicate the issue in 3.10 that is similar to our use case.

We can see the tasks being garbage collected in the [logs](https://github.com/wazuh/wazuh/pull/21694#issue-2112771714) provided in the PR description. The task created in this case is `send_entire_agent_groups_information` and we can see that each task starts its execution but only one of them finishes. This is because only one task is kept in the heap memory while the others are garbage collected. 

https://github.com/wazuh/wazuh/blob/a4b03d6ab30fa65e40ae06d823283e59df3a8322/framework/wazuh/core/cluster/master.py#L653-L679

> [!Note]
> The asynchronous operations performed inside each task are correctly awaited, it shouldn't be an issue with how we handle asynchrony.

One of the proposed solutions is to introduce a set `_pending_tasks` to keep a hard-reference to all pending tasks, but it hasn't been decided whether this will be included in a future release or at all. 

To sum up, the issue persists in 3.10 and hasn't been fixed by https://github.com/python/cpython/issues/90467. We should keep the hard reference unchanged and update the comment to avoid any confusion in the future.
